crates/hdk/src/lib.rs

```diff
@@ -1,40 +1,31 @@
 //!
-//! hApps can be installed on a device that's running a so-called conductor, Holochain's runtime. Clients can then call each zome's functions via Remote Procedure Calls (RPC).
-//! Holochain employs websocket ports for these RPCs, served by the conductor. Calls are made either from a client on localhost or from other nodes on the network.
-//! The zome function to be executed must be specified in each call. Every zome function defines the response it returns to the client.
-//! [More info on Holochain's architecture](https://developer.holochain.org/concepts/2_application_architecture)
+//! hApps can be developed using the HDK. See the [Holochain Quick Start Guide](https://developer.holochain.org/quick-start)
+//! to get started with hApp development.
 //!
-//! Low-level communication between the conductor and WASM binaries, like typing and serialization of data, is encapsulated by the HDK.
-//! Using the HDK, hApp developers can focus on their application's logic. [Learn more about WASM in Holochain.](https://github.com/holochain/holochain/blob/develop/crates/hdk/ON-WASM.md)
-//!
-//! See the [Holochain Learning Resources](https://developer.holochain.org/learning) to get started with hApp development.
+//! Functions of a hApp can be organized into reusable components. In Holochain terminology these components are called "zomes".
+//! One or multiple zomes are compiled into a WebAssembly (WASM) binary, referred to as a DNA. All of the DNAs of an application are bundled to a hApp.
+//! In short, the structure is __hApp -> DNA -> zome -> function__.
 //!
 //! # Example zomes 🍭
 //!
-//! The HDK is used in all the WASMs used to test Holochain itself.
-//! As they are used directly by tests in CI they are guaranteed to compile and work for at least the tests we define against them.
-//!
 //! There are numerous example/test WASMs on many aspects of hApp development that can be browsed
 //! [on Github](https://github.com/holochain/holochain/tree/develop/crates/test_utils/wasm/wasm_workspace).
 //!
 //! Each example WASM is a minimal demonstration of specific HDK functionality, such as generating random data, creating entries or defining validation callbacks.
 //! Some of the examples are very contrived, none are intended as production grade hApp examples, but do highlight key functionality.
 //!
-//! # Integrity zomes 📐
+//! # Zomes are separated into data model and domain logic
+//!
+//! hApps are required to produce and validate data deterministically. There's a data model and a domain logic part to each hApp. In Holochain, the
+//! data model is defined in integrity zomes and the domain logic is written in coordinator zomes.
+//!
+//! ## Integrity zomes 📐
 //!
 //! Integrity zomes describe a hApp's domain model by defining a set of entry and link types and providing a validation callback
 //! function that checks the integrity of any operations that manipulate data of those types.
```

Comment:
I don't know if we're going for an overview of the essentials or completeness here, but if completeness, maybe we should add a note about the genesis self-check:

```suggestion
//! function that checks the integrity of any operations that manipulate data of those types.
//! Additionally, a genesis self-check callback can also be provided to give basic verifiation
//! of the data that! allows an agent to join a network before they attempt to join it.
```